### PR TITLE
chore: migrate valid-describe to ts

### PR DESCRIPTION
**Pull request by M4rk9696**
Part of #256

### CODE REVIEW

File: src/rules/__tests__/valid-describe.test.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
-import { RuleTester } from 'eslint';
+import { TSESLint } from '@typescript-eslint/experimental-utils';
 import rule from '../valid-describe';
 
-const ruleTester = new RuleTester({
+const ruleTester = new TSESLint.RuleTester({
+  parser: '@typescript-eslint/parser',
```
Comment (SimenB) [inline src/rules/__tests__/valid-describe.test.ts]: You can remove this line

File: src/rules/tsUtils.ts [hunk 2]
```diff
@@ -2,10 +2,34 @@
 import { basename } from 'path';
 import { ESLintUtils } from '@typescript-eslint/experimental-utils';
 import { version } from '../../package.json';
+import {
+  AST_NODE_TYPES,
```
Comment (SimenB) [inline src/rules/tsUtils.ts]: Add these imports to line 3

File: src/rules/tsUtils.ts [hunk 3]
```diff
@@ -2,10 +2,34 @@
...
   return `${REPO_URL}/blob/v${version}/docs/rules/${ruleName}.md`;
 });
+
+enum DescribeAlias {
+  'describe',
+  'fdescribe',
+  'xdescribe',
+}
+
+export const isFunction = (node: TSESTree.Node): boolean =>
```
Comment (SimenB) [inline src/rules/tsUtils.ts]: Can this be a type guard?

`: node is TSESTree.FunctionExpression | TSESTree.ArrowFunctionExpression`

File: src/rules/valid-describe.ts [hunk 4]
```diff
@@ -73,7 +86,8 @@ export default {
             });
 
             return;
-          }
+          } else callbackFunction = callbackFunction as FunctionExpression;
```
Comment (SimenB) [inline src/rules/valid-describe.ts]: Add braces

File: src/rules/valid-describe.ts [hunk 5]
```diff
@@ -51,14 +64,14 @@ export default {
...
-          const [, callbackFunction] = node.arguments;
+          let [, callbackFunction] = node.arguments;
           if (!isString(name)) {
             context.report({
               messageId: 'firstArgumentMustBeName',
               loc: paramsLocation(node.arguments),
             });
           }
-          if (callbackFunction === undefined) {
+          if (!callbackFunction) {
```
Comment (SimenB) [inline src/rules/valid-describe.ts]: Could we do the `true` branch first, then `else`?

File: src/rules/tsUtils.ts [hunk 6]
```diff
@@ -2,10 +2,34 @@
...
+} from '@typescript-eslint/experimental-utils';
 
 const REPO_URL = 'https://github.com/jest-community/eslint-plugin-jest';
 
 export const createRule = ESLintUtils.RuleCreator(name => {
   const ruleName = basename(name, '.ts');
   return `${REPO_URL}/blob/v${version}/docs/rules/${ruleName}.md`;
 });
+
+enum DescribeAlias {
```
Comment (M4rk9696) [inline src/rules/tsUtils.ts]: @SimenB Should I replace this enum with a type.

File: src/rules/valid-describe.ts (hunk 5 above)
Comment (M4rk9696) [reply src/rules/valid-describe.ts]: sure

File: src/rules/tsUtils.ts (hunk 6 above)
Comment (SimenB) [reply src/rules/tsUtils.ts]: nah, an enum is fine 🙂

File: src/rules/valid-describe.ts [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+  node.type === AST_NODE_TYPES.TemplateLiteral;
+
+const hasParams = (node: FunctionExpression): boolean => node.params.length > 0;
+
+const paramsLocation = (
+  params: TSESTree.Expression[] | TSESTree.Parameter[],
+) => {
+  const [first] = params;
+  const last = params[params.length - 1];
+  return {
```
Comment (SimenB) [inline src/rules/valid-describe.ts]: can we do

```ts
return {
  start: first.loc.start,
  end: last.loc.end,
}
```

?

**Pull request merged**